packages/spindle-ui/src/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,158 @@
+
+type Variant =
+  | 'text'
+  | 'textWithIcon'
+  | 'headWithIcon'
+  | 'headWithIconAndCaption';
+
+interface DefaultProps {
+  children: React.ReactNode;
+}
+
+interface ListItemProps extends DefaultProps {
+  icon?: React.ReactNode;
+  onClick: () => void;
+}
+
+interface ListProps extends DefaultProps {
+  id?: string;
+  onClose: () => void;
+  open: boolean;
+  position?: 'left' | 'right';
+  triggerRef: React.RefObject<HTMLButtonElement>;
+  variant?: Variant;
+}
+
+const BLOCK_NAME = 'spui-DropdownMenu';
+const FADE_OUT_ANIMATION = 'spui-DropdownMenu-fade-out';
+
+const Caption = ({ children }: DefaultProps) => {
+  return <p className={`${BLOCK_NAME}-caption`}>{children}</p>;
+};
+
+const Frame = ({ children }: DefaultProps) => {
+  return <div className={`${BLOCK_NAME}`}>{children}</div>;
+};
+
+const List = ({
+  children,
+  id,
+  onClose,
+  open,
+  position = 'left',
+  triggerRef,
+  variant = 'text',
+}: ListProps) => {
+  const menuEl = useRef<HTMLUListElement>(null);
+  const [fadeOut, setFadeOut] = useState(false);
+
+  const onClickBody = useCallback(
+    (e) => {
+      if (!open) return;
+      if (e.composedPath().includes(triggerRef.current)) return;
+      if (
+        e.composedPath().includes(document.querySelector(`.${BLOCK_NAME}-menu`))
+      )
+        return;
+
+      setFadeOut(true);
+    },
+    [open, setFadeOut, triggerRef],
+  );
+
+  const onClickCloser = useCallback(() => {
+    setFadeOut(true);
+    triggerRef.current?.focus();
+  }, [setFadeOut, triggerRef]);
+
+  const handleKeyDown = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Escape' || e.key === 'Esc') {
+        onClickCloser();
+      }
+    },
+    [onClickCloser],
+  );
+
+  const handleAnimationEnd = useCallback(
+    (event: AnimationEvent) => {
+      if (event.animationName === FADE_OUT_ANIMATION) {
+        onClose();
+        setFadeOut(false);
+      }
+    },
+    [onClose, setFadeOut],
+  );
+
+  useEffect(() => {
+    menuEl.current?.addEventListener('animationend', handleAnimationEnd, false);
+
+    return () =>
+      menuEl.current?.removeEventListener(
+        'animationend',
+        handleAnimationEnd,
+        false,
+      );
+  }, [menuEl, handleAnimationEnd, open]);
+
+  useEffect(() => {
+    document.body.addEventListener('click', onClickBody, false);
+
+    return () => {
+      document.body.removeEventListener('click', onClickBody, false);
+    };
+  });
+
+  useEffect(() => {
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [handleKeyDown]);
+
+  return (
+    <>
+      {open && (
+        <ul
+          id={id}
+          onClick={onClickCloser}
+          className={[
+            `${BLOCK_NAME}-menu`,
+            `${BLOCK_NAME}-menu--${variant}`,
+            position === 'right' && `${BLOCK_NAME}-menu--right`,
+            fadeOut && 'is-fade-out',
+          ].join(' ')}
```

Comment:
```suggestion
          className={[
            `${BLOCK_NAME}-menu`,
            `${BLOCK_NAME}-menu--${variant}`,
            position === 'right' && `${BLOCK_NAME}-menu--right`,
            fadeOut && 'is-fade-out',
          ].filter(Boolean).join(' ')}
```

undefinedがclassNameに含まれてしまいそうなのでこうしても良いかもです